lib/scryptenc/scryptenc.c

```diff
@@ -168,9 +169,9 @@ checkparams(size_t maxmem, double maxmemfrac, double maxtime,
 
 	/* Check limits. */
 	N = (uint64_t)(1) << logN;
-	if ((memlimit / N) / r < 128)
+	if (!force && ((memlimit / N) / r < 128))
```

Comment:
Why bother calling memtouse and scryptenc_cpuperf if we're simply going to ignore them?  I'd make lines 155-162 conditional on !force and have memlimit and opslimit default to huge values otherwise.